src/System.ServiceModel.Security/tests/ServiceModel/SecurityAlgorithmSuiteTest.cs

```diff
@@ -0,0 +1,52 @@
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultEncryptionAlgorithm), "http://www.w3.org/2001/04/xmlenc#tripledes-cbc")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultEncryptionKeyDerivationLength), "192")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultSymmetricKeyWrapAlgorithm), "http://www.w3.org/2001/04/xmlenc#kw-tripledes")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultAsymmetricKeyWrapAlgorithm), "http://www.w3.org/2001/04/xmlenc#rsa-oaep-mgf1p")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultSymmetricSignatureAlgorithm), "http://www.w3.org/2000/09/xmldsig#hmac-sha1")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultAsymmetricSignatureAlgorithm), "http://www.w3.org/2000/09/xmldsig#rsa-sha1")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultSignatureKeyDerivationLength), "192")]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.DefaultSymmetricKeyLength), "192")]
+    public static void Property_SecurityAlgorithmSuite_TripleDes(string propertyName, string expectedValue)
+    {
+        // Use reflection to get the default value of the properties provided in the inline data.
+        // Verify the default values are the same as on the full framework.
+        SecurityAlgorithmSuite suite = SecurityAlgorithmSuite.TripleDes;
+        PropertyInfo info = suite.GetType().GetProperty(propertyName);
+        Assert.Equal(info.GetValue(suite).ToString(), expectedValue);
+    }
+
+    [WcfTheory]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.IsSymmetricKeyLengthSupported), true, 192)]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.IsSymmetricKeyLengthSupported), false, 257)]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.IsAsymmetricKeyLengthSupported), true, 1024)]
+    [InlineData(nameof(System.ServiceModel.Security.TripleDesSecurityAlgorithmSuite.IsAsymmetricKeyLengthSupported), false, 4097)]
+    public static void Method_SecurityAlgorithmSuite_TripleDes(string propertyName, bool expectedValue, int argVal)
+    {
+        // Use reflection to call the method with the specified arguments based on the inline data.
+        // This test verifies the called methods return true or false based on the values passed in.
+        Type objType = typeof(TripleDesSecurityAlgorithmSuite);
+        Type tripleDesSecurityAlgorithmSuite = Type.GetType(objType.AssemblyQualifiedName);
+        ConstructorInfo constructor = tripleDesSecurityAlgorithmSuite.GetConstructor(Type.EmptyTypes);
+        object desClassObject = constructor.Invoke(new object[] { });
```

Comment:
Why are you constructing an instance using reflection? Developers won't be able to do that, they will only be able to use the type returned from SecurityAlgorithmSuite.TripleDes so you just get that instance and test it.

Reply:
Fixed.